starter-code/index.js

```diff
@@ -1,11 +1,50 @@
 class SortedList {
-  constructor() {}
-  add(item) {}
-  get(pos) {}
-  max() {}
-  min() {}
-  average() {}
-  sum() {}
-};
+  constructor(items) {
+    this.items = [];
+    this.length = this.items.length;
+  }
+  add(x) {
+    this.items.push(x);
+    this.length = this.items.length;
+    this.items = this.items.sort(function(a, b) {
+      return a - b;
+    });
+  }
+  get(pos) {
+  return  this.items[pos - 1];
+  }
+  max() {
+    if (this.items.length === 0) {
+      throw new Error("EmptySortedList")
+    } else {
+     return  Math.max(...this.items);
+    }
+    
+    
+    } 
 
+  
+   min() {
+    if (this.items.length === 0) {
+      throw new Error("EmptySortedList")
+   } else {
+     return Math.min(...this.items)
+   }
+  }
+   average(){
+    if (this.items.length === 0) {
+      throw new Error("EmptySortedList")
+    } else {
```

Comment:
when calculating the average, you can call the sum function that you've already created to help keep your code dry 

```
return this.sum() / this.length;
```